### ISSUE TITLE
Cannot Pull Private Images From Docker Hub

### ISSUE THREAD

**Issue by eligundry** [Azure/docker-login#7]
I wanted to be able to pull a previously pushed image to use as a build cache in our Github actions. The following failed:

```yml
- uses: azure/docker-login@v1
  with:
    login-server: index.docker.io
    username: ${{ secrets.DOCKER_USERNAME }}
    password: ${{ secrets.DOCKER_PASSWORD }}

- name: Docker Pull Cache
  run: |
    docker pull private/repo:latest
```

Whereas, logging in with the Docker CLI tool works:

```yml
- name: Docker Login (Redneck Style)
  run: docker login --username='${{ secrets.DOCKER_USERNAME }}' --password='${{ secrets.DOCKER_PASSWORD }}'

- name: Docker Pull Cache
  run: |
    docker pull private/repo:latest
```

I suspect this is because `docker login` initiates an OAuth2 flow with scopes that allow pulling and that this action just base64 encodes an auth header that won't allow pulling for security reasons.

**Comment by thesattiraju** [comment]
Could you try removing `login-server` from your action run; letting it fall back to the default login-server url and execute again?

**Comment by eligundry** [comment]
This somehow fixes it. Thanks for the tip. I was looking at the code and I'm not certain how/why this works?

**Comment by thesattiraju** [comment]
When you use `docker login --username<> --password<>` that's the default login-url which is being set. But the url you provided is `index.docker.io` which is different from the fully qualified url. So when you're trying to pull a private image; the look up of the default url fails, so docker ends up trying to pull it without any creds.

**Comment by jromero** [comment]
This should be documented somewhere or fixed because it also resolved #4 for me.
